contracts/CoverageDataProvider.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import "./interface/ICoverageDataProvider.sol";
+import "./interface/IRegistry.sol";
+import "./interface/IVault.sol";
+import "./interface/AaveV2/IAavePriceOracle.sol";
+import "./Governable.sol";
+
+/**
+ * @title  CoverageDataProvider
+ * @author solace.fi
+ * @notice Calculates the maximum amount of cover that `Solace` protocol can sell as a coverage. 
+*/
+contract CoverageDataProvider is ICoverageDataProvider, Governable {
+    
+    /// @notice price oracle.
+    IAavePriceOracle internal _priceOracle;
+
+    /// @notice SOLACE-USDC SLP Pool address.
+    address public immutable solaceSLP = 0x9C051F8A6648a51eF324D30C235da74D060153aC;
+    /// @notice SCP address.
+    address public immutable solaceSCP = 0x501AcEe83a6f269B77c167c6701843D454E2EFA0;
+    /// @notice DAI address.
+    address public immutable dai = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    /// @notice WETH address. 
+    address public immutable weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    /// @notice USDC address.
+    address public immutable usdc = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    /// @notice WBTC address.
+    address public immutable wbtc = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+    /// @notice USDT address.
+    address public immutable usdt = 0xdAC17F958D2ee523a2206206994597C13D831ec7;
```

Comment:
can these be added and removed dynamically?

Reply:
done.